design/roadmap.md

```diff
@@ -1,179 +1,71 @@
-   * [Show errors in app files (`ks lint`)](#show-errors-in-app-files-ks-lint)
-   * [Update VSCode extension](#update-vscode-extension)
-* Under the hood
-  * [Improve `ksonnet-lib` generation](#improve-ksonnet-lib-generation)
-  * [Refactor common code into `ksonnet/client`](#refactor-common-code-into-ksonnetclient)
+## Component namespaces
 
+* Users can segregate their components into namespaces for more convenient organization.
+* Users can customize their configuration to only use a subset of their component namespaces.
 
-## Support third-party registries (beyond Github protocol)
+## Flexible environments
 
-*Planned for 0.9.0* ([#232](https://github.com/ksonnet/ksonnet/issues/232))
+* Users can user their Ksonnet project with multiple Kubernetes clusters. An environment is not directly tied to a Kubernetes cluster namespace.
 
-Currently ksonnet allows you to `ks generate` prototypes from three sources:
-* **[Standard prototypes]** From ksonnet code itself (see [`systemPrototypes.go`](https://github.com/ksonnet/ksonnet/blob/master/prototype/systemPrototypes.go))
-* **[Standard prototypes]** From the [`incubator`](https://github.com/ksonnet/parts/tree/master/incubator) registry in the `ksonnet/parts` Github repo
-* **[Custom prototypes]** From Github directories that conform to the registry spec (see [docs](https://ksonnet.io/docs/cli-reference#ks-registryks-registry-add) for `ks registry add`)
+## Registry
 
-The last option (`ks registry add`) allows users to write and share their own prototypes. *However, this feature is currently limited to Github.* Users may want more flexibility, e.g. a local-filesystem-based registry for development purposes.
+* Users can publish their parts to the Ksonnet registry.
+* Users can offer multiple versions of parts.
 
-**`ks registry add` functionality should be expanded to support other common protocols (local filesystem, S3, etc).** The only requirement thereafter will be having the proper YAML metadata files (e.g. `registry.yaml` in the root).
+## Dependencies
 
-## Easily generate third-party registries
+* Users can configure Ksonnet to use parts on the local filesystem as dependencies.
 
-*Planned for 0.9.0* ([ #234](https://github.com/ksonnet/ksonnet/issues/234))
+## Default parts
 
-Users can create custom, third-party registries, but the `ks registry add` command will only recognize them if they adhere to the registry spec. In other words, they need to have the appropriate YAML files and directory structure.
+* Support for parts based on Kubernetes types will be included with Ksonnet. These parts will act as local dependencies and be upgraded through the registry.
 
-**A new command (e.g. `ks registry create`) which autogenerates registry scaffolding files, may make this easier and less error prone.**
+## Expose environment metadata to prototypes
 
-## Lazily download and update dependencies (`ks install`)
+* Users can reference the current namespace in components.
```

Comment:
I almost want to bump this up to the section on "More flexible environments", but up to you

Reply:
actually wait, how will this work if environments can have more than one cluster namespace? how will we know what the "current namespace" is? would it need to be keyed or something?

Reply:
It will be resolved at run time. Environments can have more than one cluster namespace, and ksonnet will operate on them serially. It'll be doing operation X, Y times.